e-commerce/client/src/components/Cart.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+
+import { Link } from "react-router-dom";
+import { CartItem } from "../App";
+import { Product } from "../generated";
+
+import CartItems from "./CartItems";
+
+type CartProps = {
+  cartItems: CartItem[];
+  onAddToCart: (product: Product) => void;
+  onRemoveFromCart: (product: Product) => void;
+};
+
+const Cart: React.FunctionComponent<CartProps> = ({
+  cartItems,
+  onAddToCart,
+  onRemoveFromCart,
+}) => {
+  const getTotal = (products: CartItem[]) =>
+    products.reduce(
+      (accumulator: number, product) =>
+        accumulator + product.quantity * Number(product.product.price),
```

Comment:
If you change the price to a number field, you won't need to use `Number(` here.